src/CommunityToolkit.Maui/Mappings/SizeMappingInfo.shared.cs

```diff
@@ -0,0 +1,41 @@
+﻿using CommunityToolkit.Maui.Core.Primitives;
+
+namespace  CommunityToolkit.Maui.Mappings;
+
+/// <summary>
+/// Size mapping info about the screen diagonal-size and the category it falls into.
+/// </summary>
+public class SizeMappingInfo
```

Comment:
let's change this to be a `record struct`, it's immutable and small, also I saw a lot of it on collections, so we can take advantage of some compiler optimizations.

```suggestion
public record struct SizeMappingInfo
```